tests/providers/cncf/kubernetes/operators/test_pod.py

```diff
@@ -1009,16 +1025,50 @@ def test_wait_for_xcom_sidecar_iff_push_xcom(self, mock_await, mock_extract_xcom
         else:
             mock_await.assert_not_called()
 
-    @pytest.mark.parametrize("should_fail", [True, False])
+    @pytest.mark.parametrize(
+        "task_kwargs, should_fail, should_be_deleted",
+        [
+            ({}, False, True),
+            ({}, True, True),
+            (
+                {"is_delete_operator_pod": True, "on_finish_action": "keep_pod"},
+                False,
+                True,
+            ),  # check b/c of is_delete_operator_pod
```

Comment:
```suggestion
            ),  # check backcompat of is_delete_operator_pod
```

This might be better, took me a bit to figure out what "b/c" was.